src/jvmMain/kotlin/com/sdercolin/vlabeler/ui/dialog/EditEntryExtraDialog.kt

```diff
@@ -0,0 +1,169 @@
+    }
+
+    fun isValid(index: Int): Boolean {
+        return extraFields[index].isOptional || values[index].isNullOrBlank().not()
+    }
+
+    val isAllValid: Boolean
+        get() = values
+            .mapIndexed { index, s -> extraFields[index].isOptional || s != null }
+            .all { it }
+
+    fun cancel() {
+        finish(null)
+    }
+
+    fun trySubmit() {
+        if (isAllValid) {
+            submit()
+        }
+    }
+
+    fun submit() {
+        finish(EditEntryExtraDialogResult(entryIndex, values))
+    }
+}
+
+@Composable
+fun EditEntryExtraDialog(
+    args: EditEntryExtraDialogArgs,
+    finish: (EditEntryExtraDialogResult?) -> Unit,
+) {
+    val state = remember {
+        EditEntryExtraState(args, finish)
+    }
+
+    Column(modifier = Modifier.width(400.dp).height(400.dp)) {
+        Spacer(Modifier.height(20.dp))
+        Text(
+            text = string(Strings.EditEntryExtraDialogDescription),
+            style = MaterialTheme.typography.body2,
+            fontWeight = FontWeight.Bold,
+        )
+        Spacer(Modifier.height(25.dp))
+        ExtraContent(state)
+        Spacer(Modifier.height(25.dp))
+        Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.End) {
+            TextButton(
+                onClick = state::cancel,
+            ) {
+                Text(string(Strings.CommonCancel))
+            }
+            Spacer(Modifier.width(25.dp))
+            ConfirmButton(
+                enabled = state.isAllValid,
+                onClick = state::trySubmit,
+            )
+        }
+    }
+}
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+private fun ColumnScope.ExtraContent(
+    state: EditEntryExtraState,
+) {
+    Row(modifier = Modifier.fillMaxWidth().weight(1f)) {
+        val scrollState = rememberScrollState()
+        Column(modifier = Modifier.weight(1f).verticalScroll(scrollState)) {
+            state.extraFields.forEachIndexed { index, extraField ->
+                if (extraField.isVisible) {
+                    Row(modifier = Modifier.fillMaxWidth(), verticalAlignment = Alignment.CenterVertically) {
+                        Text(
+                            text = extraField.displayedName.get(),
+                            style = MaterialTheme.typography.body2,
+                            maxLines = 1,
+                            overflow = TextOverflow.Ellipsis,
+                        )
+                        Spacer(Modifier.width(25.dp))
+                        TextField(
+                            modifier = Modifier.weight(1f)
+                                .onKeyEvent {
+                                    if (it.isReleased(Key.Enter)) {
+                                        state.trySubmit()
+                                        true
+                                    } else {
+                                        false
+                                    }
+                                },
```

Comment:
I think it's not common to submit the whole form when entered on any item since we could have a lot here.

Reply:
I think it's useful for keyboard interactions. I assume users will not edit most of items if there are lots of extra fields and users can use tab to change focus if they don't want to submit all.